### ISSUE TITLE
Come up with a system for tracking GCAM compatibility

### ISSUE THREAD

**Issue by pralitp** [JGCRI/gcamwrapper#16]
As will continually happen, updates in GCAM such as library version updates will create incompatibilities in `gcamwrapper`.  It has already happened with TBB/Eigen and will again for `RapidXML`.

We should come up with a policy about maintaining backwards compatibility and then a strategy for how exactly to do that from a branching perspective.

**Comment by crvernon** [comment]
@pralitp Agreed, I think containerization will help with this naturally from a deployment perspective.  From a depreciation perspective, we may have to set up a pretty extensive actions suite that runs according to a cron job weekly to ensure that our versions don't break with new releases.  Versions that do break will need to have breaking libs versions pinned or set to <= the working version.

From a branching perspective, I think we can still run with (fully open to discuss and find what is best for this repo):

- `main` as the current stable version for which a release always exists.  This never gets a commit that is not a release.  Main can get a hotfix but it has to be immediately released.
- `dev` the next pending version to be released.  All feature and bug fix branches that are not hot come off of this one.
- `gh-pages` our docs that get auto-deployed along with a merge to `main` via an action.
